src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -102,86 +96,55 @@ public void run() {
             }
 
             performArchival();
```

Comment:
Since `performArchival` returns a `Future`, this should call ex. `.get()` on that `Future` so that the call blocks until the result is completed.